File: api/Rest/Product/Services/IProductService.cs
using Product.Dtos;
using System.Collections.Generic;

namespace Product.Services
{
    public interface IProductService
    {
        IList<Entity.Product> GetProducts();
        void CreateProduct(CreateProductRequest request);
        Entity.Product GetProduct(int productId);
        void UpdateProduct(UpdateProductRequest request);
        void DeleteProduct(int productId);
    }
}

File: src/modules/inventory/services/iproductService.ts
import { Promise } from "@app/common";
import { AddOrEditProductModel } from "../pages/addOrEditProduct";
export interface IProductService {
    getProducts(): Promise;
    addProduct(item: AddOrEditProductModel): Promise;
    getProduct(productId: string): Promise;
    updateProduct(product: AddOrEditProductModel): Promise;
    deleteProduct(productId: string): Promise;
}